packages/test/src/admin-utils.ts

```diff
@@ -192,8 +224,144 @@ export async function createCollectionContainingProduct(
     ],
     parentId,
   };
-  return await adminClient.query<Collection, MutationCreateCollectionArgs>(
-    CREATE_COLLECTION,
-    { input }
+  return (
+    (await adminClient.query<Collection, MutationCreateCollectionArgs>(
+      CREATE_COLLECTION,
+      { input }
+    )) as any
+  ).createCollection;
+}
+
+export async function createProduct(
+  adminClient: SimpleGraphQLClient,
+  translationFields: Omit<
+    ProductTranslationInput,
+    'customFields' | 'languageCode'
+  >
+) {
+  return (
+    (await adminClient.query<Product, MutationCreateProductArgs>(
+      CREATE_PRODUCT,
+      {
+        input: {
+          translations: [
+            {
+              languageCode: LanguageCode.En,
+              description: translationFields.description,
+              name: translationFields.name,
+              slug: translationFields.slug,
+            },
+          ],
+        },
+      }
+    )) as any
+  ).createProduct;
+}
+
+export async function assignOptionGroupsToProduct(
+  adminClient: SimpleGraphQLClient,
+  product: Product,
+  optionGroups: CreateProductOptionGroupInput[]
+): Promise<Product> {
+  const createdOptionGroups: any[] = await Promise.all(
+    optionGroups.map(async (g) => {
+      return (
+        (await adminClient.query<
+          ProductOptionGroup,
+          MutationCreateProductOptionGroupArgs
+        >(CREATE_PRODUCT_OPTION_GROUP, {
+          input: {
+            code: g.code,
+            options: g.options,
+            translations: [
+              {
+                languageCode: LanguageCode.En,
+                name: g.translations[0].name,
+              },
+            ],
+          },
+        })) as any
+      ).createProductOptionGroup;
+    })
+  );
+  // for(const optionIndex in createdOptionGroups){
+  //   createdOptionGroups[optionIndex].options=[...(
+  //     await Promise.all(await optionGroups.map(async(o)=>{
+  //       return (await adminClient.query<ProductOption, MutationCreateProductOptionArgs>(
+  //       CREATE_PRODUCT_OPTION,
+  //       { input:{
+  //         productOptionGroupId: createdOptionGroups[optionIndex].createProductOptionGroup.id,
+  //         code: o.code,
+  //         translations:[
+  //           {
+  //             languageCode: LanguageCode.En,
+  //             name: o.translations[0].name
+  //           }
+  //         ]
+  //       } }
+  //     ) as any).createProductOption}))
+  //   )]
+  // }
+  let newProduct = product;
+  for (const optionGroup of createdOptionGroups) {
+    newProduct = (
+      (await adminClient.query<Product, MutationAddOptionGroupToProductArgs>(
+        ADD_OPTION_GROUP_TO_PRODUCT,
+        {
+          optionGroupId: optionGroup.id as string,
+          productId: newProduct.id,
+        }
+      )) as any
+    ).addOptionGroupToProduct;
+  }
+  return newProduct;
+}
+
+export async function createProductVariants(
```

Comment:
Maybe rename to `createVariantsForProductOptions` or some JS doc for clarification

Reply:
ok